File: src/main/java/com/joraph/schema/Key.java
package com.joraph.schema;

import kotlin.jvm.functions.Function1;

/**
 * A key property.
 * @param <T> the property type
 */
public class Key<T, R>
		extends BaseProperty<T, R>
		implements Property<T, R> {

	/**
	 * Creates the key.
	 */
	public Key(Function1<T, R> accessor) {
		super.setPropertyChain(accessor);
	}

}

File: src/main/kotlin/com/joraph/schema/Key.kt
package com.joraph.schema

/**
 * A property that is a primary (unique) key. An entity can only
 * have a single primary key.
 */
class Key<T, R>(accessor: (T) -> R?) : BaseProperty<T, R>(accessor), Property<T, R> {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        if (!super.equals(other)) return false
        return true
    }

    override fun hashCode(): Int {
        return super.hashCode()
    }
}
